types/time.go

```diff
@@ -2206,7 +2206,13 @@ func GetFormatType(format string) (isDuration, isDate bool) {
 	}
 
 	format = skipWhiteSpace(format)
-	for token, formatRemain, succ := getFormatToken(format); len(token) != 0; format = formatRemain {
+	for {
+		var token string
+		var succ bool
+		token, format, succ = getFormatToken(format)
```

Comment:
How about defining `token` and `succ` outside this `for` loop, so that we don't need to allocate new variables for each iteration?